File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& arr) {
        int n = arr.size();
        vector<int> ans(n,0);
        int idx = 0;
        for(int i=0; i<n;i++){
            if(arr[i] != 0) ans[idx++] = arr[i];
        }
        arr = ans;

    }
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int idx = 0;
        int count =0;
        for(int i=0; i<n; i++) {
            if(nums[i] == 0) count++;
            if(nums[i] != 0) nums[idx++] =nums[i];
        }

        for(int i=idx; i<idx+count; i++){
            nums[i] = 0;
        }
    }
}